test/deprecated_test.cpp

```diff
@@ -120,7 +120,7 @@ int cpp_main(int /*argc*/, char* /*argv*/[])
     de.replace_filename("bar.foo", fs::file_status(), fs::file_status());
     BOOST_TEST(de.path() == "bar.foo");
 
-    const fs::path temp_dir(fs::current_path() / ".." / fs::unique_path("deprecated_test-%%%%-%%%%-%%%%"));
+    const fs::path temp_dir(fs::temp_directory_path() / fs::unique_path("deprecated_test-%%%%-%%%%-%%%%"));
```

Comment:
Why is this change necessary?

Reply:
> Not strictly needed. But without the change the test creates test files in your current dir, wherever that might be. And the `temp_dir` name seems to be a clue as to what was actually intended.

That's how all Filesystem tests behave. If this is to be changed, other tests need to be changed as well.

But since this is not related to modularization, please leave this out of this PR. You can create a separate PR, if you'd like this changed.